addons/website/models/website.py

```diff
@@ -1518,10 +1526,13 @@ def _search_get_details(self, search_type, order, options):
 
         :return: list of search details obtained from the `website.searchable.mixin`'s `_search_get_detail()`
         """
-        result = []
-        if search_type in ['pages', 'all']:
-            result.append(self.env['website.page']._search_get_detail(self, order, options))
-        return result
+        if search_type == 'all':
+            model_names = reduce(operator.ior, SEARCH_TYPE_MODELS.values())
+        else:
+            model_names = SEARCH_TYPE_MODELS[search_type]
+
+        return list(self.env[model_name]._search_get_detail(self, order, options)
+                    for model_name in model_names)
```

Comment:
I'm sorry about the bug, @bso-odoo , and I thank you for the explanation. I didn't expect it either.
Why not adding this instead of reverting, though, if I may ask? (Seems to work locally using your snippet)

```suggestion
                    for model_name in model_names
                    if self.env.registry.models.get(model_name))
```